subprojects/core/src/main/java/org/gradle/api/internal/TaskOutputCaching.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import org.gradle.api.Nullable;
+
+public interface TaskOutputCaching {
```

Comment:
I had `TaskCacheability` at some point. I describes if the outputs of the task are cacheable. In other words if caching the outputs of a task is enabled. How about `CachingTaskOutputs`?
The field describes characteristics of the task outputs with respect to caching. Do you think `State` is the right noun to describe characteristics? I feel that `State` is something that easily changes.